make/golang-v1.mk

```diff
@@ -150,6 +150,26 @@ else \
 fi;
 endef
 
+# golang-bump-major-version bumps the module's major version in go.mod, 
+# VERSION, swagger.yml, and non-vendored Go files.
+define golang-bump-major-version
+# bump go.mod
+$(eval $@OLD_VERSION := $(shell cat swagger.yml | grep 'version:\s[0-9].[0-9].[0-9]' | sed 's/[:|[:alpha:]|[:space:]]//g' | cut -d. -f1 ||\ 
+cat VERSION.yml | sed 's/[:|[:alpha:]|[:space:]]//g' | cut -d. -f1))
+$(eval $@NEW_VERSION := $(shell echo $$(($($@OLD_VERSION)+1))))
+sed -i '' '1s/v$(1)/v$($@NEW_VERSION)/' go.mod 
+
+# if wag, bump wag version
+test -f swagger.yml && sed -i '' 's/version: $(1).[[:digit:]].[[:digit:]]/version: $($@NEW_VERSION).0.0/' swagger.yml || true
+# if has VERSION file, bump version
+test -f VERSION && sed -i '' 's/$(1).[[:digit:]].[[:digit:]]/$($@NEW_VERSION).0.0/' VERSION || true
+
+# update imports in all *.go files
+$(eval $@_ESC_OLD_PKG :=  $(shell head -1 go.mod | sed 's/module //' | sed 's/\//\\\//g'))
+$(eval $@_ESC_NEW_PKG := $(shell head -1 go.mod | sed '1s/v$(1)/v$($@NEW_VERSION)/' | sed 's/module //' | sed 's/\//\\\//g'))
+find . -name '*.go' -not -path './vendor/*' -print0 | xargs -0 sed -i '' 's/$($@_ESC_OLD_PKG)/$($@_ESC_NEW_PKG)/'
```

Comment:
am I reading this correclty that if a repo has internal import statements (i.e. github.com/Clever/<therepo>) it now needs to make those versioned? If so that seems pretty annoying

Reply:
yep, exactly

Reply:
this was the reason for adding a function to bump major versions . Requiring people to do it manually would result in some frustrations. If one import is not updated, go modules will import the old version of the repo into itself :jijispin: